src/coordinate.py

```diff
@@ -1,15 +1,43 @@
 """Coordinate module"""
 from __future__ import annotations
-from typing import Tuple
+from typing import Tuple, Optional, Any
 
 
 class Coordinate(object):
     """Abstract each board position."""
 
-    def __init__(self, coordinate: Tuple[int, int]) -> None:
+    def __init__(self, coordinate: Any, secondCoord: Optional[int]) -> None:
         """Instantiate Coordinate with x and y values."""
-        #TODO: add overload constructor which just takes two ints instead of a tuple?
-        self._coordinate: Tuple[int, int] = coordinate
+        #self._coordinate: Tuple[int, int] = coordinate
+        if isinstance(coordinate, tuple): 
+            self.fromTuple(coordinate)
+        elif isinstance(coordinate, int) and secondCoord is not None:
```

Comment:
Same idea as the comment below - are we just going to trust our data here? If so, why check if it's `None` at all?